### PR TITLE
add benchmarks test and increase performance for metric parsing

### PR DESCRIPTION
**Pull request by salvovitale**

### CODE REVIEW

**Comment by salvovitale** [comment]
Hey guys, in order to benchmark some functions for now I made them public. In the future, I believe that we will do some refactoring and those functions will be public in their own module anyway.

File: src/prom/parser.rs [hunk 1]
```diff
@@ -141,6 +154,17 @@ fn decode_labels(labels: &str) -> HashMap<String, String> {
...
 
+pub fn decode_labels_with_rgx(labels_to_split: &str) -> HashMap<String, String>   {
+   log::debug!("decode_labels: {}", labels_to_split);
+    let regex = Regex::new(r#"(\w+)="(\w+)""#).unwrap(); // using the global "/g" mode to capture all the occurrences without stopping at the first match
+    let mut labels = HashMap::new();
+    for cap in regex.captures_iter(labels_to_split) {
+       labels.insert(cap[1].to_string(), cap[2].to_string());
+     }
+    return labels;
+    labels
```
Comment (thmshmm) [inline src/prom/parser.rs:173]: unreachable code

File: src/prom/parser.rs [hunk 2]
```diff
@@ -121,7 +121,20 @@ fn extract_type(line: &str) -> Option<String> {
...
+                let labels = line.split_at(firs_index+1).1.split_at(second_index-firs_index-1).0;
+                return Some(String::from(labels));
+            }
+            None => None
+        }
+        None => None,
+    }
+}
+
+pub fn extract_labels_with_rgx(line: &str) -> Option<String> {
```
Comment (thmshmm) [inline src/prom/parser.rs:145]: maybe add #[allow(dead_code)]

**Comment by thmshmm** [comment]
Can u also please use rust analyzer to format the document?

File: src/prom/parser.rs (hunk 1 above)
Comment (salvovitale) [reply src/prom/parser.rs:173]: done!

**Pull request merged**